src/main/java/org/broadinstitute/hellbender/tools/spark/sv/utils/SVFastqUtils.java

```diff
@@ -93,91 +94,159 @@ else if (ce == NEGATIVE.charEncoding)
     }
 
     public static final class Mapping implements Locatable {
-        
-        private final String contig;
-
-        private final int start;
 
-        private final boolean forwardStrand;
+        final List<AlignmentInterval> intervals;
 
-        private final Cigar cigar;
-
-        public Mapping(final GATKRead read) {
-            Utils.nonNull(read);
-            if (read.isUnmapped()) {
-                contig = null;
-                start = -1;
-                forwardStrand = true;
-                cigar = null;
+        public Mapping(final String str) {
+            Utils.nonNull(str);
+            if (str.equals(UNMAPPED_STR)) {
+                intervals = Collections.emptyList();
             } else {
-                contig = read.getContig();
-                start = read.getStart();
-                forwardStrand = !read.isReverseStrand();
-                cigar = read.getCigar();
+                intervals = Collections.unmodifiableList(SA_TAG_ALN_INTERVAL_SEPARATOR_PATTERN.splitAsStream(str)
+                    .filter(s -> !s.isEmpty())
+                    .map(AlignmentInterval::new)
+                    .collect(Collectors.toList()));
             }
         }
 
-        public Mapping(final String mappingDescription) {
-            Utils.nonNull(mappingDescription);
-            if (mappingDescription.equals(UNMAPPED_DESCRIPTION_STR)) {
-                contig = null;
-                start = -1;
-                forwardStrand = true;
-                cigar = null;
-            } else {
-                final Matcher matcher = MAPPING_DESCRIPTION_PATTERN.matcher(mappingDescription);
-                if (!matcher.find()) {
-                    throw new IllegalArgumentException("invalid mapping description: '" + mappingDescription + "'");
+        public Mapping(final GATKRead read) {
+            Utils.nonNull(read);
+            if (read.isUnmapped()) {
+                intervals = Collections.emptyList();
+            } else if (!read.hasAttribute(SAMTag.SA.name())) {
+                if (read.isSupplementaryAlignment()) {
+                    throw new IllegalArgumentException("a supplementary alignment read record must supply a SA tag");
```

Comment:
While true for BWA, I'm not sure it's true in general that supplementary alignments must have SA tags.  Then there's also the confusion between XA and SA tags.  If it were me, I'd leave this check out entirely.
If you do leave this test in, maybe you could find a better exception to throw than an IllegalArgumentException, which doesn't seem to me very descriptive of a malformed read.

Reply:
I think I will leave it for now indicating that this situation is unsupported.

Reply:
I'm using GATKException.